main_test.go

```diff
@@ -0,0 +1,182 @@
+			filters:        []string{"test=0"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid filter option specified, use '--filter help' for more info"),
+		},
+		{
+			testName:       "invalid filter type",
+			filters:        []string{"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=0"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid filter option specified, use '--filter help' for more info"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>0"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: >"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=a"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: a"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=4294967296"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: 4294967296"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=-1"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: -1"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=1 1"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: 1 1"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid="},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: "),
+		},
+		{
+			testName: "invalid uid",
+			filters: []string{"uid=	"},
+			expectedFilter: tracee.Filter{},
+			expectedError: errors.New("invalid UID value: 	"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=%s"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: %s"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=\t\t\t"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: \t\t\t"),
+		},
+		{
+			testName:       "invalid uid",
+			filters:        []string{"uid=0\t"},
+			expectedFilter: tracee.Filter{},
+			expectedError:  errors.New("invalid UID value: 0\t"),
+		},
+		{
+			testName: "uid=0",
+			filters:  []string{"uid=0"},
+			expectedFilter: tracee.Filter{
+				UIDFilter: tracee.UIDFilter{
+					Equal:    []uint32{0},
+					NotEqual: []uint32{},
+					Greater:  4294967295,
+					Less:     0,
```

Comment:
Shouldn't greater be 0 and less 4294967295?
I would expect that if no ">" or "<" were chosen, that these will have the default value (so they will always return true).
See my comment above about default values